security/voters.rst

```diff
@@ -163,12 +170,14 @@ would look like this::
             return true;
         }
 
-        protected function voteOnAttribute(string $attribute, mixed $subject, TokenInterface $token): bool
+        protected function voteOnAttribute(string $attribute, mixed $subject, TokenInterface $token, ?Vote $vote = null): bool
         {
             $user = $token->getUser();
+            $vote ??= new Vote();
 
             if (!$user instanceof User) {
                 // the user must be logged in; if not, deny access
+                $vote->reasons[] = 'The user is not logged in.';
```

Comment:
I would also add one entry inside the `canEdit()`

```php
    private function canEdit(Post $post, User $user, Vote $vote): bool
    {
        // this assumes that the Post object has a `getOwner()` method
        $isOwner = $user === $post->getOwner();
        if (!$isOwner) {
               $vote->reasons[] = 'You are not the owner of the Post.';
        }
        return $isOwner
    }
```

Reply:
Tweaked a bit the code but I agree with this example